### PR TITLE
Products client

### PR DESCRIPTION
**Pull request by HoolyDuck**

### CODE REVIEW

**Comment by HoolyDuck** [comment]
it's not finished and very polished yet, waiting for review

File: client/src/common/enums/api/api-routes.enum.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export {ApiRoutes} from '../../../../../shared/common/enums/enums'
```
Comment (Gr1ngoire) [inline client/src/common/enums/api/api-routes.enum.ts]: leave only shared/......

File: client/src/common/types/product/productDTO.type.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import type { ProductTagDTO } from "../productTag/productTag.types";
+
+export type ProductDTO = {
```
Comment (Gr1ngoire) [inline client/src/common/types/product/productDTO.type.ts]: Move to shared

File: client/src/common/types/productTag/productTagDTO.type.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export type ProductTagDTO = {
```
Comment (Gr1ngoire) [inline client/src/common/types/productTag/productTagDTO.type.ts]: Move to shared

File: client/src/components/Button.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
+<template>
+  <button  @click="handleClick" :class="'btn ' + bgColor">
```
Comment (Gr1ngoire) [inline client/src/components/Button.vue]: Move to @common/components

File: client/src/components/Header.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <nav>
+    <ul class="header-elements">
```
Comment (Gr1ngoire) [inline client/src/components/Header.vue]: Move to @common/components

File: client/src/router/index.ts [hunk 6]
```diff
@@ -1,14 +1,21 @@
...
       component: ExampleComponent
     }, */
-  ]
-})
 
-export default router
+    {
+      path: ClientRoutes.PRODUCTS,
+      name: "products",
+      component: () => import("@/views/products/ProductListView.vue"),
```
Comment (Gr1ngoire) [inline client/src/router/index.ts]: Import from @viwes/viwes in the beginning of the file

File: client/src/stores/productListStore.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import type { ProductDTO, ProductTagDTO } from '@/common/types/types'
+import { ProductService, ProductTagService } from '@/services/services'
+import { defineStore } from 'pinia'
+import { ref, computed, type ComputedRef } from 'vue'
+
+
+export const useProductListStore = defineStore('products', () => {
```
Comment (Gr1ngoire) [inline client/src/stores/productListStore.ts]: Store name to enum

**Review by Gr1ngoire — CHANGES_REQUESTED** [review]

**Review by Gr1ngoire — APPROVED** [review]

**Pull request merged**